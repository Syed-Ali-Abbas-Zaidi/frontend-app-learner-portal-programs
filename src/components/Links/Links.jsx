import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Icon } from '@edx/paragon';

function Links(props) {
  const { title, links, className } = props;

  const linkItems = links.map(link => (
    <li key={link.href}>
      <Icon className={['fa', 'fa-file', 'mr-2', 'text-primary']} />
      <a href={link.href} target="_blank" rel="noopener noreferrer">{link.title}</a>
    </li>
  ));

  return (
    <section className={classNames('links', className)}>
      <h3 className="mb-3">{title}</h3>
      <nav>
        <ul className="list-unstyled">
          {linkItems}
        </ul>
      </nav>
    </section>
  );
}

Links.propTypes = {
  title: PropTypes.string.isRequired,
  links: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    href: PropTypes.string.isRequired,
  })).isRequired,
  className: PropTypes.string,
};

Links.defaultProps = {
  className: undefined,
};

export default Links;
