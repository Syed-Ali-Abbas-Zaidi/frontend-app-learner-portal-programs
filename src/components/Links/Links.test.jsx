import React from 'react';
import renderer from 'react-test-renderer';

import Links from './Links';

const linksData = {
  title: 'examples',
  links: [
    {
      title: 'I am first',
      href: 'https://example1.com',
    },
    {
      title: 'I am second',
      href: 'https://example2.com',
    },
    {
      title: 'I am third',
      href: 'https://example3.com',
    },
  ],
};

describe('<Links />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create((
        <Links title={linksData.title} links={linksData.links} />
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
