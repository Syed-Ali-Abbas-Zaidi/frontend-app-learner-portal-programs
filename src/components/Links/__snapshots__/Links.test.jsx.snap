// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Links /> renders correctly 1`] = `
<section
  className="links"
>
  <h3
    className="mb-3"
  >
    examples
  </h3>
  <nav>
    <ul
      className="list-unstyled"
    >
      <li>
        <span
          aria-hidden={true}
          className="fa fa-file mr-2 text-primary"
          id="Icon1"
        />
        <a
          href="https://example1.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          I am first
        </a>
      </li>
      <li>
        <span
          aria-hidden={true}
          className="fa fa-file mr-2 text-primary"
          id="Icon1"
        />
        <a
          href="https://example2.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          I am second
        </a>
      </li>
      <li>
        <span
          aria-hidden={true}
          className="fa fa-file mr-2 text-primary"
          id="Icon1"
        />
        <a
          href="https://example3.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          I am third
        </a>
      </li>
    </ul>
  </nav>
</section>
`;
